compiler/include/expr.h

```diff
@@ -146,7 +159,11 @@ class SymExpr : public Expr {
   SymExpr* symbolSymExprsPrev;
   SymExpr* symbolSymExprsNext;
 
+  SymRename* rename;
```

Comment:
Hmm... Is there another way to do this other than adding something to `SymExpr`? My concern is that the compiler generates a gajillion SymExpr nodes and so for this particular AST node we probably do care about how big it is. A secondary concern is that renaming a Symbol seems to be something to do with `use`/`import` rather than inherent to what SymExpr is.

Can you think of a different approach that would get you the improved error messages you are looking for? (We could consider leaving this out of this PR for the moment and following up with a separate PR to improve the errors).

Reply:
The issue is that your SymExpr could be to a symbol that has been renamed multiple times, so you might have no path back to the name used in the UnresolvedSymExpr.  The case I specifically was adding for had an additional level to it - the symbol that was getting printed out wrong was part of resolving an earlier UnresolvedSymExpr, namely the module symbol in a qualified access.  E.g. we were getting `Symbol 'y' undeclared in module 'Foo'` when the user had typed `Bar.y`.  So we weren't even in a context where the original UnresolvedSymExpr was still around, it had long since left the building.

Maybe I could make a map from SymExprs to their original names?

Reply:
I think it'd help if you could move it to a separate PR. (A map might be appropriate... but is it going to be used in multiple passes? I was imagining that a 2nd try at part of scopeResolve would track these things in order to get a better error message. E.g. the 2nd pass we talked about below could make a map storing all such renamings, even if we might not want to do that in general.